From eb56bb4ee759a5888f40a5b22b3c00fb0291a229 Mon Sep 17 00:00:00 2001
From: Lukasz Bartosik <lbartosik@marvell.com>
Date: Fri, 1 Feb 2019 19:34:50 +0530
Subject: [PATCH 0037/1921] octeontx2-af: don't register CPT AF interrupts

Don't register CPT AF interrupts if CPT block is not
implemented on a given platform.

Change-Id: I3fcf69f2cae14979eb3037e8391789800d74f34e
Signed-off-by: Lukasz Bartosik <lbartosik@marvell.com>
[WK: The original patch got from Marvell sdk11.21.09]
Signed-off-by: Wenlin Kang <wenlin.kang@windriver.com>
---
 drivers/net/ethernet/marvell/octeontx2/af/rvu.c | 8 +++++---
 1 file changed, 5 insertions(+), 3 deletions(-)

diff --git a/drivers/net/ethernet/marvell/octeontx2/af/rvu.c b/drivers/net/ethernet/marvell/octeontx2/af/rvu.c
index 9e1195f54297..ed4bef545624 100644
--- a/drivers/net/ethernet/marvell/octeontx2/af/rvu.c
+++ b/drivers/net/ethernet/marvell/octeontx2/af/rvu.c
@@ -2582,9 +2582,11 @@ static int rvu_probe(struct pci_dev *pdev, const struct pci_device_id *id)
 	if (err)
 		goto err_flr;
 
-	err = rvu_cpt_register_interrupts(rvu);
-	if (err)
-		goto err_irq;
+	if (is_block_implemented(rvu->hw, BLKADDR_CPT0)) {
+		err = rvu_cpt_register_interrupts(rvu);
+		if (err)
+			goto err_irq;
+	}
 
 	err = rvu_policy_init(rvu);
 	if (err)
-- 
2.31.1

